services/workflows-service/src/storage/get-file-storage-manager.ts

```diff
@@ -0,0 +1,71 @@
+import { diskStorage } from 'multer';
+import { getFileName } from '@/storage/get-file-name';
+import { z } from 'zod';
+import multerS3 from 'multer-s3';
+import { S3Client, S3ClientConfig, GetObjectCommand } from '@aws-sdk/client-s3';
+import * as tmp from 'tmp';
+import * as fs from 'fs';
+
+import { Readable } from 'stream';
+export type TLocalFile = string;
+
+const isS3BucketConfigured = () => {
+  return process.env.AWS_S3_BUCKET_KEY;
+};
+
+const generateAwsConfig = (): S3ClientConfig => {
+  return {
+    region: z.string().parse(process.env.AWS_REGION),
+    credentials: {
+      accessKeyId: z.string().parse(process.env.AWS_S3_BUCKET_KEY),
+      secretAccessKey: z.string().parse(process.env.AWS_S3_BUCKET_SECRET),
+    },
+  };
+};
+
+export const fetchDefaultBucketName = () => {
+  return z.string().parse(process.env.AWS_S3_BUCKET_NAME);
+};
+
+export const manageFileByProvider = () => {
+  if (isS3BucketConfigured()) {
+    return multerS3({
+      s3: new S3Client(generateAwsConfig()),
```

Comment:
Thoughts about having a single S3 client exported? I see this file has multiple instances of `S3Client`.

Reply:
This client is configured and dynamicly passed to the "interface".
as in the future we'll have Fido AWS config and multiple "clients" talking to eachother (moving files from one to another"